src/components/RangeDatePicker/BaseRangeDatePicker.tsx

```diff
@@ -0,0 +1,285 @@
+import React from 'react'
+import { CSSProperties, Ref, useEffect, useRef, useState } from 'react'
+
+import { ExternalStyles, focusBoxShadow, Theme, useStyles } from '../../styles'
+import { composeRefs } from '../../util/react'
+import { DateInput } from '../DateField'
+import { Icons } from '../Icon'
+import { InputWrapper } from '../TextField/InputWrapper'
+
+export interface Period {
+  startDate?: Date
+  finalDate?: Date
+}
+
+export interface BaseRangeDatePickerProps {
+  /**
+   * Prop used to set an initial period.
+   */
+  value?: Period
+
+  /**
+   *  Props to control limited periods
+   */
+  minDate?: Date
+  maxDate?: Date
+
+  /**
+   * Prop to disable the date field.
+   */
+  disabled?: boolean
+
+  /**
+   * Prop to set an icon in date field button.
+   * When this prop is not declared, date field will be
+   * shown without a icon.
+   */
+  icon?: Icons
+
+  /**
+   *  Prop to reference an invalid state.
+   */
+  invalid?: boolean
+
+  /**
+   *  Prop for receive external styles.
+   */
+  style?: ExternalStyles
+
+  /**
+   *  Props to set placeholder to inputs
+   */
+  startPlaceholder?: string
+  finalPlaceholder?: string
```

Comment:
It is better to write a separate doc/comment for each attribute, because the doc is associated just to the attribute it is immediately above (in that case, just the `startPlaceholder` attribute has the documentation associated)